db/external_sst_file_ingestion_job.cc

```diff
@@ -851,6 +870,12 @@ Status ExternalSstFileIngestionJob::GetIngestedFileInfo(
       return Status::Corruption("Corrupted key in external file. ",
                                 pik_status.getState());
     }
+    // For SST files from live db, we only support ingestion if all keys have
+    // zero sequence number and there are no duplicate keys. However, without
+    // scanning the input file, we can only check their boundary keys. To
+    // support ingesting such files, our code base needs to be able to handle
+    // duplicate keys with the same user key and sequence number. The block
+    // look up path assumes there is no duplicate key which also has to change.
```

Comment:
Can we scan the whole thing? The checks for zero at endpoints were redundant before as the table properties implied the seqnos would all be zero. But in case of `from_live_db` they're critical for correctness as we don't know where the file is coming from.

We could put smallest/largest seqno in a table property later to optimize it. I saw this was discussed already below

Reply:
Yes, added scanning validations when allow_db_generated_files is true.